src/nvim/ex_getln.c

```diff
@@ -3470,6 +3479,9 @@ void redrawcmd(void)
   msg_no_more = FALSE;
 
   set_cmdspos_cursor();
+  if (extra_char != NUL) {
+    putcmdline(extra_char, true);
```

Comment:
We could change the signature of `putcmdline()` to `void putcmdline(int c, bool shift)` in `ex_getln.c` as well.